src/mds/CDir.cc

```diff
@@ -1409,7 +1409,7 @@ CDir::fnode_ptr CDir::project_fnode(const MutationRef& mut)
   return pf;
 }
 
-void CDir::pop_and_dirty_projected_fnode(LogSegment *ls, const MutationRef& mut)
+void CDir::pop_and_dirty_projected_fnode(LogSegmentRef& ls, const MutationRef& mut)
```

Comment:
@batrick, I need more details here..what's the benefit of passing the `LogSegmentRef&` as const.
Is it for ensuring that the SP counter will not be changed in this code path (asking as it's already passes with '&' to avoid inc/dec) ?  

> (here and elsewhere)

do you imply to pass with const between functions in others paths too?

Reply:
> @batrick, I need more details here..what's the benefit of passing the LogSegmentRef& as const.

It is generally a code smell in C++ to pass a `non-const` reference, especially for a code base like Ceph trying to conform to Google's code style guidelines.

> Is it for ensuring that the SP counter will not be changed in this code path (asking as it's already passes with '&' to avoid inc/dec) ?

Yes.

> do you imply to pass with const between functions in others paths too?

Yes.